lib/utils/sdk.ts

```diff
@@ -24,71 +24,46 @@ import type {
 
 /**
  * Get a fleet object, disambiguating the fleet identifier which may be a
- * a fleet slug, name or numeric database ID (as a string).
+ * a fleet slug or name.
  * TODO: add support for fleet UUIDs.
  */
 export async function getApplication(
 	sdk: BalenaSDK,
-	nameOrSlugOrId: string | number,
+	nameOrSlug: string,
 	options?: PineOptions<Application>,
 ): Promise<Application> {
-	const { looksLikeFleetSlug, looksLikeInteger } = await import('./validation');
-	if (
-		typeof nameOrSlugOrId === 'string' &&
-		looksLikeFleetSlug(nameOrSlugOrId)
-	) {
+	const { looksLikeFleetSlug } = await import('./validation');
+	if (typeof nameOrSlug === 'string' && looksLikeFleetSlug(nameOrSlug)) {
 		return await sdk.models.application.getDirectlyAccessible(
-			nameOrSlugOrId,
+			nameOrSlug,
 			options,
 		);
 	}
-	if (typeof nameOrSlugOrId === 'number' || looksLikeInteger(nameOrSlugOrId)) {
-		try {
-			// Test for existence of app with this numerical ID
-			return await sdk.models.application.getDirectlyAccessible(
-				Number(nameOrSlugOrId),
-				options,
-			);
-		} catch (e) {
-			if (typeof nameOrSlugOrId === 'number') {
-				throw e;
-			}
-			const { instanceOf } = await import('../errors');
-			const { BalenaApplicationNotFound } = await import('balena-errors');
-			if (!instanceOf(e, BalenaApplicationNotFound)) {
-				throw e;
-			}
-			// App with this numerical ID not found, but there may be an app with this numerical name.
-		}
-	}
-	// Not a slug and not a numeric database ID: must be an app name.
+	// Not a slug: must be an app name.
 	// TODO: revisit this logic when we add support for fleet UUIDs.
 	return await sdk.models.application.getAppByName(
-		nameOrSlugOrId,
+		nameOrSlug,
 		options,
 		'directly_accessible',
 	);
```

Comment:
We could still allow this to accept numbers, but I see that it was a forcing function surfacing more commands that might be using IDs.

Let's just drop the unnecessary string check and move the special case handling first:
```suggestion
	const { looksLikeFleetSlug } = await import('./validation');
	if (!looksLikeFleetSlug(nameOrSlug)) {
		// Not a slug: must be an app name.
		// TODO: revisit this logic when we add support for fleet UUIDs.
		return await sdk.models.application.getAppByName(
			nameOrSlug,
			options,
			'directly_accessible',
		);
	}

	return await sdk.models.application.getDirectlyAccessible(
		nameOrSlug,
		options,
	);
```

Applies to `getFleetSlug()` as well.